clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -801,6 +843,43 @@ class SyclKernelFieldChecker
   bool IsInvalid = false;
   DiagnosticsEngine &Diag;
 
+  // Check whether the object is bit-wise copyable
+  bool copyableToKernel(const FieldDecl *FD, const QualType &FieldTy) {
```

Comment:
This is actually a 'check' type function, and thus should be 'checkCopyableToKernel'.  With that, please invert the 'bool' (false is successful).  Then, all of the Diagnostics.report, followed by a return changes to "return Diagnostics.Report", since that is implicitly convertable to bool.

Reply:
OK